src/effects/RealtimeEffectStateUI.cpp

```diff
@@ -0,0 +1,266 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  RealtimeEffectStateUI.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#include "RealtimeEffectStateUI.h"
+
+#include <cassert>
+// wx/weakref.h misses this include
+#include <type_traits>
+#include <wx/weakref.h>
+
+#include "EffectUI.h"
+#include "RealtimeEffectState.h"
+
+#include "EffectManager.h"
+#include "ProjectWindow.h"
+#include "Track.h"
+
+namespace
+{
+const RealtimeEffectState::Extensions::RegisteredFactory realtimeEffectStateUIFactory { [](auto&)
+   { return std::make_unique<RealtimeEffectStateUI>(); }
+};
+} // namespace
+
+
+// A shared state, that manages the UI
+class RealtimeEffectStateUI::SharedState final
+{
+public:
+   ~SharedState()
+   {
+      Hide();
+   }
+
+   void SetActiveState(RealtimeEffectState& state)
+   {
+      mCurrentState = state.shared_from_this();
```

Comment:
At first I feared a leaky `shared_ptr` cycle but then I saw `mCurrentState` is a `weak_ptr`.  Good, so use `weak_from_this()` instead to make that more obvious.